Sources/CoreFoundation/CFRunLoop.c

```diff
@@ -72,7 +72,7 @@ extern bool _dispatch_runloop_root_queue_perform_4CF(dispatch_queue_t queue);
 
 #if TARGET_OS_MAC
 typedef mach_port_t dispatch_runloop_handle_t;
-#elif defined(__linux__) || defined(__FreeBSD__)
+#elif defined(__linux__)
 typedef int dispatch_runloop_handle_t;
```

Comment:
I believe that there is a type mismatch here which could truncate the value.

Reply:
Removing `|| defined(__FreeBSD__)` here makes `dispatch_runloop_handle_t` type alias to 'uint64_t' (line 80), which fallback to the generic unix implementation.

Currently, although the runloop handle type is defined for FreeBSD, it is not doing anything useful.